Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -1622,7 +1618,12 @@ JSValue Interpreter::executeEval(EvalExecutable* eval, JSValue thisValue, JSScop
     // Execute the code:
     throwScope.release();
     ASSERT(jitCode == eval->generatedJITCode().ptr());
-    return JSValue::decode(vmEntryToJavaScript(jitCode->addressForCall(), &vm, &protoCallFrame));
+    // eval code only uses scope at the beginning (op_enter).
+    // We can replace the current scope for the subsequent run.
+    callee->setScope(vm, scope);
+    EncodedJSValue result = vmEntryToJavaScript(jitCode->addressForCall(), &vm, &protoCallFrame);
+    callee->setScope(vm, globalObject->globalScope());
```

Comment:
Why we'd like to reset the scope with `globalObject->globalScope()` after? Since previously the case for
```
callee = JSCallee::create(vm, globalObject, scope);
```
we didn't reset that with `globalObject->globalScope()`.

Reply:
It is possible that we will keep the old scope from JSGlobalObject since it holds this shared evalScope, and in this case, it may keep reachable objects alive even after this scope is no longer used. So we explicitly clear this here to the global's default scope to avoid memory leaking.